src/geofeatures/GeoFeatures.js

```diff
@@ -0,0 +1,181 @@
+import Api from '../api/Api';
+import { isValidUid } from '../uid';
+
+/**
+ * @class GeoFeatures
+ * @description
+ * GeoFeatures class used to request organisation unit coordinates from the Web API.
+ */
+class GeoFeatures {
+    /**
+     * @constructor
+     *
+     * @param {Array} orgUnits Organisation units (UID, level, group, user org. unit) to include in the response.
+     * @param {String} displayName The name property to display (NAME|SHORTNAME).
+     */
+    constructor(orgUnits = [], displayName) {
+        this.orgUnits = orgUnits;
+        this.displayName = displayName;
+    }
+
+    /**
+     * @method byOrgUnit
+     *
+     * @param {Array} orgUnits Organisation units (UID, level, group, user org. unit) to include in the response.
+     *
+     * @returns {GeoFeatures}
+     */
+    byOrgUnit(orgUnits) {
+        if (!orgUnits) {
+            return this;
+        }
+
+        const orgUnitsArray = [].concat(orgUnits);
+
+        if (!orgUnitsArray.every(GeoFeatures.isValidOrgUnit)) {
+            throw new Error(`Invalid organisation unit: ${orgUnits}`);
+        }
+
+        return new GeoFeatures(orgUnitsArray, this.displayName);
+    }
+
+    /**
+     * @method byOrgUnit
+     *
+     * @param {String} displayName The name property to display (NAME|SHORTNAME).
+     *
+     * @returns {GeoFeatures}
+     */
+    displayProperty(displayName) {
+        if (!displayName) {
+            return this;
+        }
+
+        if (!GeoFeatures.isValidDisplayName(displayName)) {
+            throw new Error(`Invalid display property: ${displayName}`);
+        }
+
+        return new GeoFeatures(this.orgUnits, displayName);
+    }
+
+    /**
+     * @method getAll
+     *
+     * @param {Object} params Extra URL params to pass to the Web API endpoint.
+     *
+     * @returns {Promise} with an array of geofeatures.
+     */
+    getAll(params = {}) {
+        const api = Api.getApi();
+        const urlParams = params;
+
+        if (this.orgUnits.length) {
+            urlParams.ou = `ou:${this.orgUnits.join(';')}`;
+        }
+
+        if (this.displayName) {
+            urlParams.displayProperty = this.displayName;
+        }
+
+        return api.get('geoFeatures', urlParams);
+    }
+
+    /**
+     * @method isValidOrgUnit
+     * @static
+     *
+     * @returns {boolean} True if the org. unit is valid
+     *
+     * @description
+     * Checks if the org. unit is valid (UID, level, group, user org. unit)
+     */
+    static isValidOrgUnit(orgUnit) {
+        return (
+            isValidUid(orgUnit) ||
```

Comment:
Is it enough that the UID is valid?

Reply:
Yes, if an org unit in the hierarchy is selected - it's only an UID. If not and UID it passes on to the checks below.